packages/paste-core/components/inline-control-group/src/InlineControlGroup.tsx

```diff
@@ -58,18 +58,34 @@ const InlineControlGroup = React.forwardRef<HTMLFieldSetElement, InlineControlGr
         </Label>
         {helpText && <HelpText marginTop="space0">{helpText}</HelpText>}
         <Box element={`${element}_SET`} marginRight="space20">
-          {React.Children.map(children, (child) => {
-            return (
-              <Box
-                element={`${element}_FIELD`}
-                display={orientation === 'horizontal' ? 'inline-block' : 'block'}
-                marginTop="space40"
-                marginRight={orientation === 'horizontal' ? 'space70' : null}
-              >
-                {child}
-              </Box>
-            );
-          })}
+          <Box display={fieldStyleProps && orientation === 'horizontal' ? 'inline-flex' : 'block'}>
```

Comment:
hacky solution but this is checking whether it's being used in a horizontal visual picker and adding display="inline-flex" if so, so that all horizontal visual pickers have equal heights even if the content differs

Reply:
Yup, definitely dislike this, but at this stage I can't think of a better solution thats' in the scope of the ticket. We'll need to pencil in a big refactor of checkboxes and radios at some point in the future